pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -2363,11 +2394,13 @@ func TestEnsureExternalDNSDeployment(t *testing.T) {
 					},
 				},
 			},
+			errExpected: true,
```

Comment:
Why `Doesn't exist` test case became an error case?
If the deployment doesn't exist it has to be created and match the desired deployment. You just need to add `testSecret()` into existing objects as for the other test cases.

Reply:
Done here 6f4d092af25d1c2fc93cb34347d5e03f794cf83d